### PR TITLE
Ambient light

### PR DESCRIPTION
**Pull request by GuoLei1990**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
Merge `AmbientLight` and `EnvironmentMapLight` ino `AmbientLight`.
Add `AmbientLight` for `Scene`.

### CODE REVIEW

File: packages/core/src/lighting/AmbientLight.ts [hunk 1]
```diff
@@ -1,56 +1,139 @@
...
+   * Specular reflection intensity.
+   */
+  get specularIntensity(): number {
+    return this._specularIntensity;
+  }
+
+  set specularIntensity(value: number) {
+    this._specularIntensity = value;
+    this._scene.shaderData.setFloat(AmbientLight._specularIntensityProperty, value);
+  }
```
Comment (zhuxudong) [inline packages/core/src/lighting/AmbientLight.ts:106]: Do we still need `specularColor` in ambientLight and shader ?

Comment (GuoLei1990) [reply packages/core/src/lighting/AmbientLight.ts:106]: `specularColor` can come form Materil, for example BlinnPhongMaterial. 
In `AmbientLight` must use `specularTexture`.

Comment (GuoLei1990) [reply packages/core/src/lighting/AmbientLight.ts:106]: done

File: packages/core/src/lighting/AmbientLight.ts [hunk 2]
```diff
@@ -1,56 +1,139 @@
...
   }
 
-  set color(value: Color) {
-    this._color = value;
-    this.scene.shaderData.setColor(AmbientLight._colorProperty, this.lightColor);
+  set diffuseSolidColor(value: Color) {
+    if (value !== this._diffuseSolidColor) {
+      value.cloneTo(this._diffuseSolidColor);
+      this._scene.shaderData.setColor(AmbientLight._diffuseColorProperty, value);
+    }
```
[the remark this answers is not in the document]
Comment (GuoLei1990) [reply packages/core/src/lighting/AmbientLight.ts:59]: done

**Review by zhuxudong — APPROVED** [review]
+1

**Pull request merged**